Ran YAMLlint on all yaml files and resolved linting issues (fixes #1297)

nonbinaryfrog opened the pull request:
The only file that I couldn't quite resolve is ./sql/mozfun/norm/product_info/metadata.yaml because the specific issue is that some lines of code are too long and these specific lines are in the form of a table. I don't know how to make it shorter without completely deleting some of the information on the table.

jklukas commented:
> The only file that I couldn't quite resolve is ./sql/mozfun/norm/product_info/metadata.yaml because the specific issue is that some lines of code are too long and these specific lines are in the form of a table. I don't know how to make it shorter without completely deleting some of the information on the table.

It would be acceptable to move the table (starting at "  The mappings are as follows:...") from `./sql/mozfun/norm/product_info/metadata.yaml` into `./sql/mozfun/norm/product_info/README.md`. The docs generation inserts the content of the metadata's description into README before rendering, so I believe that strategy would preserve the rendered output. If we go that direction, we should also update the commentary in the other files in that directory so that they consistently mention the table as existing in the README.md.

jklukas commented on tests/sql/moz-fx-data-shared-prod/account_ecosystem_derived/ecosystem_user_id_lookup_v1/test_script/telemetry_stable.account_ecosystem_v4.yaml:
```diff
@@ -1,4 +1,5 @@
-- submission_timestamp: 2020-04-01 22:09:03.243074 UTC
```
These test files need to present a yaml list of documents. In this case, it's only a single document, but needs to be preceeded by a `-` to make sure it's a list rather than having all these fields at the top level. The next file has the same issue.

wlach commented:
@nonbinaryfrog can you rebase on master? I believe that should fix the "dags up to date problem". 

99% sure the "build" issues are just due to the fact that you're making a pull request from an untrusted fork. @jklukas @acmiyaguchi can you confirm?

jklukas commented:
> 99% sure the "build" issues are just due to the fact that you're making a pull request from an untrusted fork

Indeed. @nonbinaryfrog - If you are able to run the tests locally, please do so. It may be prohibitive, though, if you don't already have a GCP account set up for the tests to run queries in.

We can force integration tests to run here by pushing the fork contents to a branch on the main repo. I just did so via:

```
git-push-fork-to-upstream-branch origin nonbinaryfrog:master
```

so that should give us a sense of any tests that were affected by changes here.

We'll need to have an owner here run that each time to trigger the `build` to run again. It's an unfortunate bump in the workflow, but necessary to avoid attacks that could spill GCP credentials to logs.

jklukas commented:
It looks like the build output is showing real errors. I've verified locally that several of these are failing. Could be due to errors in yaml. I will try to rule out other possibilities later today.

nonbinaryfrog commented:
> @nonbinaryfrog can you rebase on master? I believe that should fix the "dags up to date problem".

just tried to rebase on master but it looks like nothing's changed with the dags problem.

acmiyaguchi commented:
@nonbinaryfrog Are you sure you pushed your rebased branch to the PR here? I don't see any changes since the last commit in the hour.

acmiyaguchi commented on .circleci/config.yml:
```diff
@@ -109,95 +117,101 @@ jobs:
+          name: Build and deploy docs
+          command: |
+           PATH="venv/bin:$PATH" script/generate_docs
+            --output_dir=generated_docs/
```
This is going to break the script because it doesn't have a proper line break. Is it not possible to keep this on one line? Otherwise I suggest this:

```bash
PATH="venv/bin:$PATH" script/generate_docs \
     --output_dir=generated_docs/
```

nonbinaryfrog replied:
I had to input line breaks to fix another error, apparently lines can only be 80 characters long. If this is a better way to introduce line breaks, I can do that

acmiyaguchi replied:
Yes, please introduce the backslash at the end of the first line, or bash will treat the second line as it's own command. This will lead to subtle bugs.

jklukas commented:
It looks like at least several of the build errors are due to `query_params.yaml` losing a leading `-`. These files should look like:

```
---
- name: submission_date
   type: DATE
   value: 2019-01-01
```

rather than:

```

---
name: submission_date
type: DATE
value: 2019-01-01
```

acmiyaguchi commented on .circleci/config.yml:
```diff
@@ -109,95 +117,101 @@ jobs:
+          command:
+            echo 'IMAGE="${CIRCLE_PROJECT_USERNAME+$CIRCLE_PROJECT_USERNAME/}
+            ${CIRCLE_PROJECT_REPONAME
+            :-bigquery-etl}:${CIRCLE_TAG:-latest}"' > $BASH_ENV
```
Please try to leave this on the same line. This particular formatting looks like it'll break variable substitution.

nonbinaryfrog replied:
If I leave it on the same line, it will cause a 'line too long' error. Is there another way to resolve this?

acmiyaguchi replied:
Maybe this won't break, if a newline doesn't add any new characters. I would much prefer the following though, because it the variable substitution should be considered one statement.

            ${CIRCLE_PROJECT_REPONAME:-bigquery-etl}:${CIRCLE_TAG:-latest}
            > $BASH_ENV

acmiyaguchi commented on .circleci/config.yml:
```diff
@@ -109,95 +117,101 @@ jobs:
+          name: Deploy to Dockerhub
+          command: |
+           echo "${DOCKER_PASS:?}" | docker login -u
+            "${DOCKER_USER:?}" --password-stdin
```
I believe this will break the script too.

nonbinaryfrog replied:
Will adding the backslash work on this as well?

acmiyaguchi replied:
Yes, but I would add the backslash after the pipe (i.e. `|`).

acmiyaguchi commented on sql/moz-fx-data-shared-prod/activity_stream_bi/impression_stats_by_experiment_v1/metadata.yaml:
```diff
@@ -1,5 +1,8 @@
 friendly_name: Impression Stats By Experiment
-description: Representation of tile impression statistics, clustered by experiment_id to allow efficient analysis of individual experiments
+description: Representation of tile impression statistics,
+ clustered by experiment_id to allow efficient analysis of
```
Does the line break introduce any new characters?

nonbinaryfrog replied:
I don't believe so. Why?

acmiyaguchi replied:
Some of the tests are breaking lines using the same style, I just wanted to understand the behavior. It looks like trailing whitespaces can be introduced here, which could actually lead to the field being different than it was before the change.

nonbinaryfrog commented:
> It looks like at least several of the build errors are due to `query_params.yaml` losing a leading `-`. These files should look like:
> 
> ```
> ---
> - name: submission_date
>   type: DATE
>   value: 2019-01-01
> ```
> 
> rather than:
> 
> ```
> 
> ---
> name: submission_date
> type: DATE
> value: 2019-01-01
> ```

which directory is that in?

nonbinaryfrog commented:
> @nonbinaryfrog Are you sure you pushed your rebased branch to the PR here? I don't see any changes since the last commit in the hour.

Let me try again.

acmiyaguchi commented on tests/sql/moz-fx-data-shared-prod/search_derived/mobile_search_clients_last_seen_v1/test_day_bit_shifting/expect.yaml:
```diff
@@ -23,17 +23,23 @@
+        search_with_ads: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+        ad_click: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+  days_seen_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+   AAAAAAAAAAAAAAAAAABw==
```
There are a few tests that are failing because the newline here introduces a trailing whitespace:

```
    'days_tagged_searched_bytes': 'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA '
                                  'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAABA==',
```

Please trim any trailing whitespace.

nonbinaryfrog replied:
Trimming the trailing space causes errors in a number of files

acmiyaguchi replied:
With `allow-non-breakable-inline-mappings: true`, you should be able to leave this in one piece.

acmiyaguchi commented on sql/mozfun/norm/fenix_build_to_date/metadata.yaml:
```diff
@@ -1,8 +1,12 @@
-  Fenix originally used an [8-digit app_build format](https://github.com/mozilla-mobile/fenix/blob/c7283447/automation/gradle/versionCode.gradle#L12-L16)
+  Fenix originally used an [8-digit app_build format]
+  (https://github.com/mozilla-mobile/fenix/blob/c7283447/automation
+  /gradle/versionCode.gradle#L12-L16)
```
I don't think we should be breaking URLs across multiple lines. Could you enable the following setting and undo any breaks in URLs?

```
rules:
  line-length:
    max: 80
    allow-non-breakable-words: true
    allow-non-breakable-inline-mappings: true
```

nonbinaryfrog replied:
where do I find that?

acmiyaguchi replied:
See https://yamllint.readthedocs.io/en/stable/configuration.html#configuration

```yaml
extends: default

rules:
  line-length:
    max: 80
    allow-non-breakable-words: true
    allow-non-breakable-inline-mappings: true
```

nonbinaryfrog replied:
I'm still getting the "line too long" error, even when I enable this setting somehow

acmiyaguchi replied:
Did you modify the yamllint statement to read from the configuration file?

nonbinaryfrog replied:
No I'd been using the custom config without config file route. I'll try it the other way but first, do I add the enabled setting to the config file before that?

acmiyaguchi replied:
Yes, add this setting and use the yamllint command pointing to the configuration file. It should allow for less line breaks then there are currently.

acmiyaguchi commented on .circleci/config.yml:
```diff
@@ -1,32 +1,36 @@
+          keys:
+            # when lock files change, use increasingly general
+            # patterns to restore cache
+            - &cache_key python-packages-v1
+              -{{ .Branch }}-{{ checksum "requirements.in" }}
+              -{{ checksum "requirements.txt" }}
+            - python-packages-v1-{{ .Branch }}
+              -{{ checksum "requirements.in" }}-
+            - python-packages-v1-{{ .Branch }}-
+            - python-packages-v1-
```
Undo the line breaks here.

acmiyaguchi commented on .circleci/config.yml:
```diff
@@ -109,95 +117,102 @@ jobs:
+      - run:
+          name: Determine docker image name
+          command:
+            echo 'IMAGE="${CIRCLE_PROJECT_USERNAME+$CIRCLE_PROJECT_USERNAME/} \
```
Make sure there is no whitespace after the backslash. Even better, leave this on one line.

acmiyaguchi commented on .circleci/config.yml:
```diff
@@ -109,95 +117,102 @@ jobs:
+          name: Deploy to Dockerhub
+          command: |
+           echo "${DOCKER_PASS:?}" \
+            | docker login -u "${DOCKER_USER:?}" --password-stdin \
```
Do not put a line continuation on this line, because these are suppose to be two statements.

```suggestion
            echo "${DOCKER_PASS:?}" | \
                docker login -u "${DOCKER_USER:?}" --password-stdin
```

acmiyaguchi replied:
This is still an issue.

acmiyaguchi commented on sql/moz-fx-data-shared-prod/internet_outages/global_outages_v1/metadata.yaml:
```diff
@@ -1,30 +1,60 @@
+    - `proportion_timeout`: the proportion of users that had their connection
+    timeout while uploading telemetry ([after 90s, in Firefox Desktop]
+    (https://searchfox.org/mozilla-central/rev/fa2df28a49883612bd7af4dacd80c
+    dfedcccd2f6/toolkit/components/telemetry/app/TelemetrySend.jsm#81)).
```
undo line break

acmiyaguchi commented on sql/moz-fx-data-shared-prod/telemetry_derived/italy_covid19_outage_v1/metadata.yaml:
```diff
@@ -1,25 +1,55 @@
+    - `proportion_timeout`: the proportion of users that had their
+    connection timeout while uploading telemetry ([after 90s, in Firefox]
+    (https://searchfox.org/mozilla-central/rev/fa2df28a49883612bd7af4dacd80cd
+    fedcccd2f6/toolkit/components/telemetry/app/TelemetrySend.jsm#81)).
```
undo line break

acmiyaguchi commented on sql/moz-fx-data-shared-prod/udf/boolean_histogram_to_boolean/metadata.yaml:
```diff
@@ -1,3 +1,7 @@
+  "true" bucket, or FALSE if it has a value in the "false" bucket,
+  or NULL otherwise.
+  https://github.com/mozilla/telemetry-batch-view/blob/ea0733c/src
+  /main/scala/com/mozilla/telemetry/utils/MainPing.scala#L309-L317
```
undo line break

acmiyaguchi commented on sql/moz-fx-data-shared-prod/udf/days_seen_bytes_to_rfm/metadata.yaml:
```diff
@@ -1,4 +1,5 @@
+---
+description: Return the frequency, recency, and T from a BYTE array, as
+  defined in https://lifetimes.readthedocs.io/en/latest/Quickstart.html#
+  the-shape-of-your-data RFM refers to Recency, Frequency, and Monetary value.
```
undo line break

acmiyaguchi commented on sql/moz-fx-data-shared-prod/udf/fenix_build_to_datetime/metadata.yaml:
```diff
@@ -1,8 +1,11 @@
 
-  Fenix originally used an [8-digit app_build format](https://github.com/mozilla-mobile/fenix/blob/c7283447/automation/gradle/versionCode.gradle#L12-L16)
+  Fenix originally used an [8-digit app_build format](https://github.com
+  /mozilla-mobile/fenix/blob/c7283447/automation/gradle/versionCode.gradle#L12-L16)
```
undo line break

acmiyaguchi commented on sql/moz-fx-data-shared-prod/udf/fenix_build_to_datetime/metadata.yaml:
```diff
@@ -13,11 +16,13 @@ description: |
 
-  Newer builds use a [10-digit format](https://github.com/mozilla-mobile/fenix/blob/e6ee13dc/buildSrc/src/main/java/Config.kt#L55-L112)
+  Newer builds use a [10-digit format](https://github.com/mozilla-mobile
+  /fenix/blob/e6ee13dc/buildSrc/src/main/java/Config.kt#L55-L112)
```
undo line break

acmiyaguchi commented on sql/moz-fx-data-shared-prod/udf/histogram_max_key_with_nonzero_value/metadata.yaml:
```diff
@@ -1,2 +1,5 @@
+---
+description: Find the largest numeric bucket that contains a
+  value greater than zero.  https://github.com/mozilla/telemetry-batch-view
+  /blob/ea0733c/src/main/scala/com/mozilla/telemetry/utils/MainPing.scala#L253-L266
```
undo linebreak

acmiyaguchi commented on sql/moz-fx-data-shared-prod/udf/parse_desktop_telemetry_uri/metadata.yaml:
```diff
@@ -1,5 +1,8 @@
+  /http_edge_spec.html#special-handling-for-firefox-desktop-telemetry
+  the format is /submit/telemetry/docId/docType/appName/appVersion
+  /appUpdateChannel/appBuildID e.g. /submit/telemetry/ce39b608-f595-4c69-b6a6-
+  f7a436604648/main/Firefox/61.0a1/nightly/20180328030202
```
undo line breaks

acmiyaguchi replied:
This is still broken.

acmiyaguchi commented on sql/moz-fx-data-shared-prod/udf_js/main_summary_active_addons/metadata.yaml:
```diff
@@ -1,6 +1,11 @@
+  or boolean for foreignInstall and userDisabled; string or number for
+  version.  https://github.com/mozilla/telemetry-batch-view/blob/ea
+  0733c00df191501b39d2c4e2ece3fe703a0ef3/src/main/scala/com/mozilla
+  /telemetry/views/MainSummaryView.scala#L422-L449'
```
undo line break

acmiyaguchi commented on sql/moz-fx-data-shared-prod/udf_js/main_summary_addon_scalars/metadata.yaml:
```diff
@@ -1,3 +1,7 @@
+description: Parse scalars from payload.processes.dynamic into map columns
+ for each value type.
+ https://github.com/mozilla/telemetry-batch-view/blob/ea0733c00df191501b39d2
+ c4e2ece3fe703a0ef3/src/main/scala/com/mozilla/telemetry/utils/MainPing
```
undo line break

acmiyaguchi commented on sql/moz-fx-data-shared-prod/udf_js/main_summary_disabled_addons/metadata.yaml:
```diff
@@ -1,4 +1,9 @@
+  because they are legacy). We need this as addonDetails may contain both
+  disabled and active addons.
+  https://github.com/mozilla/telemetry-batch-view/blob/ea0733c00df191501b
+  39d2c4e2ece3fe703a0ef3/src/main/scala/com/mozilla/telemetry/views
```
undo line break

acmiyaguchi commented on sql/mozfun/norm/fenix_app_info/metadata.yaml:
```diff
@@ -1,39 +1,50 @@
+  (Firefox vs. Firefox Preview) and channel names
+   (nightly, beta, or release) has changed over time; see the
+  [spreadsheet of naming history for Mozilla's mobile browsers]
+  (https://docs.google.com/spreadsheets/d/18PzkzZxdpFl23__-CIO735NumYDqu7jHpqllo0sBbPA/edit#gid=0).
```
This is not valid markdown, make sure there is no whitespace between `](`

acmiyaguchi commented on sql/mozfun/norm/fenix_build_to_date/metadata.yaml:
```diff
@@ -1,8 +1,11 @@
 
-  Fenix originally used an [8-digit app_build format](https://github.com/mozilla-mobile/fenix/blob/c7283447/automation/gradle/versionCode.gradle#L12-L16)
+  Fenix originally used an [8-digit app_build format]
+  (https://github.com/mozilla-mobile/fenix/blob/c7283447/automation/gradle/versionCode.gradle#L12-L16)
```
This is not valid markdown

acmiyaguchi commented on sql/mozfun/norm/fenix_build_to_date/metadata.yaml:
```diff
@@ -13,11 +16,13 @@ description: |
 
-  Newer builds use a [10-digit format](https://github.com/mozilla-mobile/fenix/blob/e6ee13dc/buildSrc/src/main/java/Config.kt#L55-L112)
+  Newer builds use a [10-digit format]
+  (https://github.com/mozilla-mobile/fenix/blob/e6ee13dc/buildSrc/src/main/java/Config.kt#L55-L112)
```
This is not valid markdown

acmiyaguchi commented on sql/mozfun/norm/os/metadata.yaml:
```diff
@@ -1,6 +1,11 @@
 
-  This is a reimplementation of [logic used in the data pipeline](https://github.com/mozilla/gcp-ingestion/blob/a6928fb089f1652856147c4605df715f327edfcd/ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/NormalizeAttributes.java#L52-L74) to populate `normalized_os`.
+  This is a reimplementation of [logic used in the data pipeline]
+  (https://github.com/mozilla/gcp-ingestion/blob
```
This is not valid markdown

acmiyaguchi commented on sql/mozfun/norm/os/metadata.yaml:
```diff
@@ -1,6 +1,11 @@
-  This is a reimplementation of [logic used in the data pipeline](https://github.com/mozilla/gcp-ingestion/blob/a6928fb089f1652856147c4605df715f327edfcd/ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/NormalizeAttributes.java#L52-L74) to populate `normalized_os`.
+  This is a reimplementation of [logic used in the data pipeline]
+  (https://github.com/mozilla/gcp-ingestion/blob
+  /a6928fb089f1652856147c4605df715f327edfcd/ingestion-beam/
```
undo line breaks

acmiyaguchi commented on tests/sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_users_daily_v1/test_aggregation/query_params.yaml:
```diff
@@ -1,3 +1,4 @@
-  type: DATE
-  value: 2019-01-01
+---
+- - name: submission_date
```
This looks wrong (two hyphens?)

acmiyaguchi commented on tests/sql/moz-fx-data-shared-prod/org_mozilla_firefox_derived/events_daily_v1/test_single_day/expect.yaml:
```diff
@@ -1,14 +1,12 @@
+- submission_date: "2020-01-03"
+  client_id: "client-1"
+  events: "\"\U00000001\U00000001,\U00000001\U00000001\U00000002,
+  \"\"\U00000001,"
```
undo linebreak

acmiyaguchi commented on tests/sql/moz-fx-data-shared-prod/org_mozilla_firefox_derived/events_daily_v1/test_single_day/query_params.yaml:
```diff
@@ -1,3 +1,4 @@
-  type: DATE
-  value: 2020-01-03 
+---
+name: submission_date
```
this is not a list

acmiyaguchi commented on tests/sql/moz-fx-data-shared-prod/search_derived/mobile_search_clients_last_seen_v1/test_day_bit_shifting/expect.yaml:
```diff
@@ -23,17 +23,23 @@
+        search_with_ads: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+        ad_click: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+  days_seen_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+   AAAAAAAAAAAAAAAAAABw==
```
undo these line breaks

acmiyaguchi commented on tests/sql/moz-fx-data-shared-prod/search_derived/mobile_search_clients_last_seen_v1/test_day_bit_shifting/mobile_search_clients_last_seen_v1.yaml:
```diff
@@ -23,17 +23,23 @@
+        tagged_searches: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+        search_with_ads: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+        ad_click: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+  days_seen_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+   AAAAAAAAAAAAAAAAAw==
+  days_searched_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+   AAAAAAAAAAAAAAAAAAAAAg==
+  days_tagged_searched_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+   AAAAAAAAAAAAAAAAAAAAAAAAAAAAg==
+  days_searched_with_ads_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+   AAAAAAAAAAAAAAAAAAAAAAAAAAAAAg==
+  days_clicked_ads_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+   AAAAAAAAAAAAAAAAAAAAAAg==
+  days_created_profile_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+   AAAAAAAAAAAAAAAAAAAAAAAAAyA==
```
undo these line breaks

acmiyaguchi replied:
This is still broken

acmiyaguchi commented on tests/sql/moz-fx-data-shared-prod/search_derived/mobile_search_clients_last_seen_v1/test_day_bit_shifting/mobile_search_clients_last_seen_v1.yaml:
```diff
@@ -47,15 +53,21 @@
+        tagged_searches: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2]
+        search_with_ads: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2]
+        ad_click: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2]
+  days_seen_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+   AAAAAAAAAAAAAAAAAAw==
+  days_searched_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+   AAAAAAAAAAAAAAAAAAAAAg==
+  days_tagged_searched_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+   AAAAAAAAAAAAAAAAAAAAAAAAAAAAAg==
+  days_searched_with_ads_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+   AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAg==
+  days_clicked_ads_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+   AAAAAAAAAAAAAAAAAAAAAAAg==
+  days_created_profile_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+   AAAAAAAAAAAAAAAAAAAAAAAAAAyA==
```
undo these line breaks

acmiyaguchi replied:
This is still broken

acmiyaguchi commented on tests/sql/moz-fx-data-shared-prod/telemetry_derived/clients_daily_histogram_aggregates_parent_v1/test_aggregation/moz-fx-data-shared-prod.telemetry_stable.main_v4.yaml:
```diff
@@ -3,24 +3,16 @@
   normalized_os: first
   normalized_channel: nightly
-  document_id: avg_1
+  # document_id: avg_1
```
undo this comment

acmiyaguchi commented on tests/sql/moz-fx-data-shared-prod/telemetry_derived/clients_daily_keyed_boolean_aggregates_v1/test_aggregation/moz-fx-data-shared-prod.telemetry_stable.main_v4.yaml:
```diff
@@ -6,7 +6,7 @@
   normalized_os: first
   normalized_channel: nightly
-  document_id: avg_1
+  # document_id: avg_1
```
undo this comment

acmiyaguchi commented on tests/sql/moz-fx-data-shared-prod/telemetry_derived/clients_daily_keyed_histogram_aggregates_v1/test_aggregation/moz-fx-data-shared-prod.telemetry_stable.main_v4.yaml:
```diff
@@ -3,32 +3,24 @@
   normalized_os: first
   normalized_channel: nightly
-  document_id: avg_1
+  # document_id: avg_1
```
undo this comment

acmiyaguchi commented on tests/sql/moz-fx-data-shared-prod/telemetry_derived/clients_daily_keyed_scalar_aggregates_v1/test_aggregation/moz-fx-data-shared-prod.telemetry_stable.main_v4.yaml:
```diff
@@ -6,7 +6,7 @@
   normalized_os: first
   normalized_channel: nightly
-  document_id: avg_1
+  # document_id: avg_1
```
undo this comment

acmiyaguchi commented on .circleci/config.yml:
```diff
@@ -109,95 +115,100 @@ jobs:
+      - run:
+          name: Determine docker image name
+          command:
+            echo 'IMAGE="${CIRCLE_PROJECT_USERNAME+$CIRCLE_PROJECT_USERNAME/} ${CIRCLE_PROJECT_REPONAME:-bigquery-etl}:${CIRCLE_TAG:-latest}"' > $BASH_ENV
```
```suggestion
            echo 'IMAGE="${CIRCLE_PROJECT_USERNAME+$CIRCLE_PROJECT_USERNAME/}${CIRCLE_PROJECT_REPONAME:-bigquery-etl}:${CIRCLE_TAG:-latest}"' > $BASH_ENV
```

acmiyaguchi commented on .yamllint.yaml:
```diff
@@ -0,0 +1,7 @@
+    max: 80
+    allow-non-breakable-words: true
+    allow-non-breakable-inline-mappings: true
+    
```
extra space at the end of the file (maybe run yamllint on the yaml config)

acmiyaguchi commented on sql/moz-fx-data-shared-prod/internet_outages/global_outages_v1/metadata.yaml:
```diff
@@ -1,30 +1,59 @@
+    telemetry for a reason that was not listed in the other cases.
+    - `proportion_timeout`: the proportion of users that had their connection
+    timeout while uploading telemetry ([after 90s, in Firefox Desktop]
+    (https://searchfox.org/mozilla-central/rev/fa2df28a49883612bd7af4dacd80cdfedcccd2f6/toolkit/components/telemetry/app/TelemetrySend.jsm#81)).
```
This is invalid markdown due to whitespace

acmiyaguchi commented on sql/moz-fx-data-shared-prod/telemetry_derived/italy_covid19_outage_v1/metadata.yaml:
```diff
@@ -1,25 +1,54 @@
+    to send telemetry for a reason that was not listed in the other cases.
+    - `proportion_timeout`: the proportion of users that had their
+    connection timeout while uploading telemetry ([after 90s, in Firefox]
+    (https://searchfox.org/mozilla-central/rev/fa2df28a49883612bd7af4dacd80cdfedcccd2f6/toolkit/components/telemetry/app/TelemetrySend.jsm#81)).
```
invalid markdown

acmiyaguchi commented on sql/moz-fx-data-shared-prod/udf/bitmask_range/metadata.yaml:
```diff
@@ -1,6 +1,10 @@
+  is the number of bits to include in the mask.  The arguments were chosen
+  to match the semantics of the SUBSTR function;
+  see https://cloud.google.com/bigquery/docs/reference/standard-sql
+  /functions-and-operators#substr
```
broken link

acmiyaguchi commented on .yamllint.yaml:
```diff
@@ -0,0 +1,6 @@
+  line-length:
+    max: 80
+    allow-non-breakable-words: true
+    allow-non-breakable-inline-mappings: true
```
This is missing a newline now.

acmiyaguchi reviewed (commented):
It looks like circleci is not running properly, try validating the config before pushing again: https://circleci.com/docs/2.0/local-cli/#validate-a-circleci-config

acmiyaguchi commented on .circleci/config.yml:
```diff
@@ -1,32 +1,37 @@
+            # when lock files change, use increasingly general
+            # patterns to restore cache
+            - &cache_key 
+            - python-packages-v1-{{ .Branch }}-{{ checksum "requirements.in" }}-{{ checksum "requirements.txt" }}
```
This is breaking CI

```suggestion
            python-packages-v1-{{ .Branch }}-{{ checksum "requirements.in" }}-{{ checksum "requirements.txt" }}
```

acmiyaguchi commented on .circleci/config.yml:
```diff
@@ -1,32 +1,39 @@
+      - run:
+          name: Yamllint Test
+          command: |
+          venv/bin/pip install yamllint
```
You probably don't need this line since it's included in the requirements.txt

acmiyaguchi commented on .circleci/config.yml:
```diff
@@ -39,11 +41,19 @@ jobs:
+      - checkout
+      - run: 
+          name: Verify that yaml files are linted for errors
+          command: PATH="venv/bin:$PATH" yamllint -c .yamllint.yaml
```
Doing this is more complicated than before, because the environment doesn't have the requirements set up as a separate stage. It also has to be added to the workflows. I'd suggest reverting back to putting it before the pytest and linters.

nonbinaryfrog replied:
ok will do!

acmiyaguchi commented on tests/sql/moz-fx-data-shared-prod/search_derived/mobile_search_clients_last_seen_v1/test_day_bit_shifting/expect.yaml:
```diff
@@ -23,18 +23,18 @@
   days_searched_with_ads_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABA==
   days_clicked_ads_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABA==
   days_created_profile_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABkA==
-- <<: *base
```
This should be a list.

acmiyaguchi replied:
I think this may be indented incorrectly, and the row is being read as a a list with 1 item instead of a list with 2 items.

acmiyaguchi commented on tests/sql/moz-fx-data-shared-prod/search_derived/mobile_search_clients_last_seen_v1/test_day_bit_shifting/expect.yaml:
```diff
@@ -23,18 +23,18 @@
   days_clicked_ads_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABA==
   days_created_profile_bytes: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABkA==
-- <<: *base
+  - - <<: *base
```
```suggestion
- <<: *base
```

acmiyaguchi commented on sql/moz-fx-data-shared-prod/account_ecosystem_derived/fxa_logging_users_daily_v1/metadata.yaml:
```diff
@@ -12,5 +14,7 @@ scheduling:
+    - ['moz-fx-data-shared-prod', 'firefox_accounts_stable',
+       'account_ecosystem_v1']
+    - ['moz-fx-data-shared-prod', 'account_ecosystem_derived',
+       ecosystem_user_id_lookup_v1']
```
Put these back so they're on one line.

acmiyaguchi commented on sql/moz-fx-data-shared-prod/account_ecosystem_derived/ecosystem_user_id_lookup_v1/metadata.yaml:
```diff
@@ -14,8 +15,9 @@ labels:
-  referenced_tables: [['moz-fx-data-shared-prod', 'firefox_accounts_stable', 'account_ecosystem_v1']]
+  depends_on_past: true
+  referenced_tables: [['moz-fx-data-shared-prod', 'firefox_accounts_stable',
+                       'account_ecosystem_v1']]
```
This might actually be the problematic file, put this one on a single line instead.

nonbinaryfrog replied:
When I put this on a single line and run it locally, I get a line-length error

nonbinaryfrog replied:
Should I use the disable line-length rule comment on top of it?

acmiyaguchi replied:
What about adopting the style of the other files in similar to this and putting it like:

```
referenced_tables:
  - ["moz-fx-data-shared-prod", ...]
```

nonbinaryfrog replied:
No, that still causes an error. 

This way runs fine though.

`    - [
        [
          "moz-fx-data-shared-prod",
          "firefox_accounts_stable",
          "account_ecosystem_v1",
        ],
      ]`

acmiyaguchi commented on sql/moz-fx-data-shared-prod/account_ecosystem_derived/fxa_logging_users_daily_v1/metadata.yaml:
```diff
@@ -12,5 +14,5 @@ scheduling:
-    - ['moz-fx-data-shared-prod', 'firefox_accounts_stable', 'account_ecosystem_v1']
-    - ['moz-fx-data-shared-prod', 'account_ecosystem_derived', 'ecosystem_user_id_lookup_v1']
+    - ["moz-fx-data-shared-prod","firefox_accounts_stable","account_ecosystem_v1",]
+    - ["moz-fx-data-shared-prod","account_ecosystem_derived",ecosystem_user_id_lookup_v1',]
```
This is missing proper quotes

```suggestion
    - ["moz-fx-data-shared-prod","account_ecosystem_derived", "ecosystem_user_id_lookup_v1",]
```

acmiyaguchi reviewed (approved):
Thanks for the back and forth to get this working with all of the CI here. This looks good to me.
